fox_news_scraper/extract_fn.py

```diff
@@ -0,0 +1,100 @@
+"""A file to extract information from various Fox News RSS feeds."""
+
+import feedparser
+import grequests
+from bs4 import BeautifulSoup
+
+
+def fetch_rss_feed(feed_url: str) -> feedparser.FeedParserDict:
+    """Fetches and parsers the RSS feed from the provided URL."""
+
+    feed = feedparser.parse(feed_url)
+    return feed
+
+
+def get_article_content(article_url: str) -> str:
+    """Scrapes the full content of an article from a given URL."""
+
+    try:
+        response = grequests.get(article_url)
+        if response.status_code != 200:
+            return f"Couldn't connect to article, status_code: {response.status_code}"
+
+        soup = BeautifulSoup(response.content, "html.parser")
+        article_body = soup.find("div", class_="article-body")
+
+        if article_body:
+            return article_body.get_text(separator=" ", strip=True)
+
+        return "Full content not found or unable to parse."
+
+    except Exception as e:
+        return f"Failed to fetch full content from {article_url}: {e}"
+
+
+def parse_feed_entries(feed: feedparser.FeedParserDict) -> list[dict]:
+    """Parses the entries of a feed and extracts relevant fields."""
+
+    entries = []
+    article_requests = (grequests.get(entry.link) for entry in feed.entries)
+
+    responses = grequests.map(article_requests)
+
+    for entry, response in zip(feed.entries, responses):
+        if response and response.status_code == 200:
+            soup = BeautifulSoup(response.content, "html.parser")
```

Comment:
I think line 45 to 47 could be refactored into a new function. maybe repurpose the get_article_content function to do this.